libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -590,40 +570,26 @@ protected void onPostExecute(OAuth2.TokenEndpointResponse tr) {
                 mgr.getAdminPermsManager().setPrefs(id.customPermissions, account);
             }
 
+            // Save the user account
+            addAccount(account);
+
             // Screen lock required by mobile policy.
             if (id.screenLockTimeout > 0) {
 
                 // Stores the mobile policy for the org.
                 final PasscodeManager passcodeManager = mgr.getPasscodeManager();
                 passcodeManager.storeMobilePolicyForOrg(account, id.screenLockTimeout * 1000 * 60, id.pinLength);
                 passcodeManager.setTimeoutMs(id.screenLockTimeout * 1000 * 60);
-                boolean changeRequired = passcodeManager.setMinPasscodeLength((Activity) getContext(), id.pinLength);
```

Comment:
I think we still need this for the multi-user scenario. The reason why we don't consider login complete until passcode creation is complete is because there's logic here to match the new policy against the old one (if an extra user logs in) and pick the best combination of the two (or three or more) and pick a new length and timeout. If this happens, we need to kick off the change passcode flow, which can only be initialized from here - the passcode manager has no knowledge of this. Also, it needs to be enforced right after this user logs in (new security policy).

Reply:
The solution is to have the passcode manager know that a passcode change is required (see https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1814/commits/b9ff5a9d52abd707de49299d6a0f2d2889d96e1b).
No longer showing passcode activity from random places (like OAuthWebViewHelper or PasscodeManager's setMinPasscodeLength) instead capturing the state and letting the next SalesforceActivity that resumes and that asks the PasscodeManager if app should be locked trigger the showing of the PasscodeActivity.